### ISSUE TITLE
JDK: not using global menu bar?

### ISSUE THREAD

**Issue by mszoek** [ravynsoft/ravynos#131]
The app I tried presented its menus within the app window instead of using the system menu bar.

**Comment by probonopd** [comment]
Should you ever find out how to do this please be sure to ping me or even better send a pull request documenting this highly guarded piece of knowledge at https://github.com/helloSystem/docs/blob/main/developer/menu.md.
Thanks!

**Comment by mszoek** [comment]
> Should you ever find out how to do this please be sure to ping me or even better send a pull request documenting this highly guarded piece of knowledge at https://github.com/helloSystem/docs/blob/main/developer/menu.md.

Of course! I will let you know how I solve it.

Global menus work for GTK and Qt apps because those libraries implemented detection and support for global menubar systems like com.canonical.AppMenu. I suspect we need to build this into the JDK as well so that all Java apps transparently use the DBus menu if one is available. I might be able to leverage the support that openjdk already has for macOS here.

(I see this request has been open for a lonnnnng time: https://bugs.openjdk.java.net/browse/JDK-8091321. Maybe they will take a PR for it.)

> Knowledge about the gloabal menu must be improved.

:100:

> To this day, I could not find out what really is missing for Chrome and Firefox to use the global menu on FreeBSD.

Well.. I'm packaging Firefox next so maybe I will find out :)

**Comment by probonopd** [comment]
> I suspect we need to build this into the JDK as well

Isn't this what https://launchpad.net/jayatana is supposed to do?

> Well.. I'm packaging Firefox next so maybe I will find out :)

Cool. Maybe this helps:

https://forum.manjaro.org/t/howto-enabling-kde-global-menu-support-in-mozilla-firefox-and-thunderbird/7405

**Comment by mszoek** [comment]
> Isn't this what https://launchpad.net/jayatana is supposed to do?

Yes. I've read that it works but is slow. It only works with Java Swing, and it requires being added to the command line or app properties to enable using the global menu bar. So it's not very transparent, unlike Qt/GTK/AppKit's support.
